### PR TITLE
GTFS Schedule (except type2) dbt migration

### PR DESCRIPTION
**Pull request by lauriemerrell**
## Describe your changes 

This PR migrates the `gtfs_views_staging`, `gtfs_views`, and `gtfs_schedule` DAGs / BigQuery datasets to dbt.

It combines the following PRs, previously reviewed: 

* #1364 - `gtfs_views_staging`
* #1373 - `gtfs_schedule`
* #1381 - `gtfs_views`

Each of those PRs explains unique considerations for those DAGs/datasets.

In addition, on this branch directly (i.e., 🆕 in this PR, not previously reviewed):
* **Adds SQLFluff linter pre-commit hooks**
   * I ran the linter + manually ran `python3 -m sqlfluff fix --rules L010,L019,L003,L022,L023,L036 --ignore parsing,templating --dialect bigquery ./models` on the `warehouse` directory to get everything in order on this branch
   * Per conversation with @atvaccaro and @mjumbewu I am having it just run the linter by default (no fixer by default) in the pre-commit hook because SQLFluff was more opinionated than I wanted it to be
* Migrates the `rt_views.gtfs_rt_fact_daily_feeds` and `rt_views.gtfs_rt_fact_daily_validation_errors` tasks to dbt (since they depend on GTFS views -- didn't want to break the DAG)
* Updates Airflow documentation to account for deleted DAGs

Fixes https://github.com/cal-itp/data-infra/issues/1258

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) -- deletes DAGs
- [x] Documentation
   * Updated dbt docs
   * Updated some Airflow docs

## How has this been tested?
Confirmed that the basic dbt commands work:
- [x] `dbt run`
- [x] `dbt test`
- [x] `dbt docs generate`

Note that `validation_code_descriptions` and `validation_notice_fields` currently do not work in `run`/`test`. This is because they rely on sources that are being moved in this PR and don't exist yet in prod so they can't reference the correct locations. Because these tables are isolated in the pipeline (they have no dependencies up or downstream) and no substantive changes were made to the SQL, I feel confident merging this and letting it run in prod. If there are issues they should be straightforward to address.

- [x] Ran `INFORMATION_SCHEMA` checks to confirm no tables/columns dropped -- see below for details
- [x] Ran some spot checks (`FULL OUTER JOIN`s on tables that were more heavily affected by linter checks -- `gtfs_schedule_fact_daily_pathways`, `dim_metric_date`, and `gtfs_schedule_fact_daily_pathways`, confirmed that table contents are identical between `laurie_views` and prod `views`)
- [ ] confirm that `gtfs_loader.validation` tasks run
^ The moved `gtfs_loader.validation_code_descriptions` task runs without any problems. However the `gtfs_loader.validation_notice_fields` task keeps timing out; I think it's basically just an incredibly inefficient task (see #1395) and it's just overwhelming my local Airflow. Because it runs just fine in production Airflow right now (again, literally no SQL has changed, I just moved it to an upstream DAG to treat it appropriately as a `source` for dbt), I am comfortable with this.  

<details>
<summary> SQL used to check that no tables/columns were dropped from GTFS views </summary>

```
SELECT DISTINCT t1.table_name, t2.table_name
FROM `cal-itp-data-infra-staging.laurie_views.INFORMATION_SCHEMA.COLUMNS` as t1
FULL OUTER JOIN `cal-itp-data-infra.views.INFORMATION_SCHEMA.COLUMNS` as t2
    USING(table_name, column_name)
WHERE 
    (t1.table_name IS NULL 
    OR t1.column_name IS NULL
    OR t2.table_name IS NULL
    OR t2.column_name IS NULL)
    AND t2.table_name NOT LIKE "airtable_%"
    AND t2.table_name NOT LIKE "amplitude_%"
ORDER BY t1.table_name, t2.table_name
```

^ this query does return results, I went through manually and checked that they are all other domains (like Payments) or deprecated (there are a bunch of tables that appear to have been one-offs from 2021 that are not updated programmatically)
</details>

<details>
<summary> SQL used to check that no tables/columns were dropped from GTFS schedule dataset </summary>

```
SELECT DISTINCT t1.table_name, t1.column_name, t2.table_name, t2.column_name
FROM `cal-itp-data-infra-staging.laurie_gtfs_schedule.INFORMATION_SCHEMA.COLUMNS` as t1
FULL OUTER JOIN `cal-itp-data-infra.gtfs_schedule.INFORMATION_SCHEMA.COLUMNS` as t2
    USING(table_name, column_name)
WHERE 
    (t1.table_name IS NULL 
    OR t1.column_name IS NULL
    OR t2.table_name IS NULL
    OR t2.column_name IS NULL)
ORDER BY t1.table_name, t2.table_name
```

^ this query does return results, I went through manually and checked that they are deprecated (two tables: `calitp_status` and `validation_report` -- see PR #1162 for more)
</details>

<details>
<summary> SQL used to check that no tables/columns were dropped from GTFS views staging dataset </summary>

```
SELECT DISTINCT t1.table_name, t1.column_name, t2.table_name, t2.column_name
FROM `cal-itp-data-infra-staging.laurie_gtfs_views_staging.INFORMATION_SCHEMA.COLUMNS` as t1
FULL OUTER JOIN `cal-itp-data-infra.gtfs_views_staging.INFORMATION_SCHEMA.COLUMNS` as t2
    USING(table_name, column_name)
WHERE 
    (t1.table_name IS NULL 
    OR t1.column_name IS NULL
    OR t2.table_name IS NULL
    OR t2.column_name IS NULL)
ORDER BY t1.table_name, t2.table_name
```

^ this query returns no results
</details>

### CODE REVIEW

**Review by atvaccaro — APPROVED** [review]
lgtm! pretty sure I've already reviewed most/all of this?

had a couple small comments

File: warehouse/models/gtfs_schedule_latest_only/_gtfs_schedule_latest_only.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+models:
+  - name: agency
+    description: |
+      This table is a latest-only cut of the cleaned GTFS data.
+      It contains current-only data for agencies we are currently scraping.
+      Each row is a cleaned row from an agency.txt file.
+      Definitions for the original GTFS fields are available at:
+      https://gtfs.org/reference/static#agencytxt.
+    columns:
+      - name: calitp_itp_id
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/gtfs_schedule_latest_only/_gtfs_schedule_latest_only.yml]: Resolved in 43f0ec08896b9b6abce514c4d42c1aa325ffd796 and 14485aa2985c228a790e30a8f0a51d9bb09898cc

File: warehouse/models/gtfs_schedule_latest_only/agency.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+latest_only_source as (
+    SELECT *
+    FROM {{ ref('calitp_feeds') }}
+),
+{{ get_latest_schedule_data(
+    latest_only_source = 'latest_only_source',
+    table_name = 'agency',
+    clean_table_name = 'agency_clean') }}
+
+SELECT * FROM agency
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/gtfs_schedule_latest_only/agency.sql]: Resolved in 0317607786c55e8bdfdae70c8f5b70ee741d5113

File: warehouse/macros/get_latest_schedule_data.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{% macro get_latest_schedule_data(latest_only_source, table_name, clean_table_name) %}
```
Comment (atvaccaro) [inline warehouse/macros/get_latest_schedule_data.sql:1]: this is mostly a nitpick but usually macros produce a full query that you wrap in a subquery or a CTE; so the call ends up looking like

```
with latest as ( {{ get_latest_schedule_data(...) }} )

select * from latest
```

**Review by atvaccaro — APPROVED** [review]
nitpick on that macro, it seems fairly specific so not a big deal

File: warehouse/macros/get_latest_schedule_data.sql (hunk 3 above)
Comment (lauriemerrell) [reply warehouse/macros/get_latest_schedule_data.sql:1]: Resolved 19387b1dd54c68c7e20e573f2b6cd7150b1e18e0

**Review by atvaccaro — APPROVED** [review]
:shipit:

**Comment by lauriemerrell** [comment]
🚨 Holding off on merge until @mjumbewu and/or @j-meelah are able to review from a Payments perspective

**Review by j-meelah — APPROVED** [review]

File: warehouse/models/gtfs_schedule_latest_only/_gtfs_schedule_latest_only.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjumbewu) [reply warehouse/models/gtfs_schedule_latest_only/_gtfs_schedule_latest_only.yml]: Ooh, I didn't even know those were a thing :student:

File: warehouse/dbt_project.yml [hunk 4]
```diff
@@ -34,5 +34,14 @@ models:
...
+    rt_views:
+      schema: views
+
+    gtfs_views_staging:
+      schema: gtfs_views_staging
+
+    gtfs_schedule_latest_only:
+      schema: gtfs_schedule
+
+    gtfs_views:
```
Comment (mjumbewu) [inline warehouse/dbt_project.yml:46]: Can we alphabetize these model folders? As we get more of them in here, it's going to make the yml easier to scan.

Comment (lauriemerrell) [reply warehouse/dbt_project.yml]: Resolved in cc95af9138f52ee6343d6988f456fd5980b17242

**Review by mjumbewu — COMMENTED** [review]
Thank you so much for plowing through on this! I just had a couple of small things, in addition to the following...

**question:** Where do we actually run the DBT models? Is there a new DAG or DAG task somewhere that I'm missing? Also, same question for the tests.

File: warehouse/macros/scd_join.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{% macro scd_join(tbl_a, tbl_b, using_cols, created_col="calitp_extracted_at",
+    deleted_col="calitp_deleted_at", join_type="JOIN",
+    sel_left_cols=None,
+    sel_right_cols=None) %}
+
+    {% set left_cols %}
+    {% if sel_left_cols==None %}
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/macros/scd_join.sql:7]: I had actually tried to convert this to `is None` initially and it yelled at me 😢 

Just tried again to confirm and here's the error:

[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/macros/scd_join.sql:7]: So, I guess this is some templating shenanigans? But unfortunately leaving as won't do for now

**Comment by lauriemerrell** [comment]
@mjumbewu re:
> question: Where do we actually run the DBT models? Is there a new DAG or DAG task somewhere that I'm missing? Also, same question for the tests.

Both `dbt run` and `dbt test` are in here (already running in prod): https://github.com/cal-itp/data-infra/tree/main/airflow/dags/transform_warehouse -- `dbt run` was added in #1272. Scheduled to run at *noon* UTC so it's offset from all the midnight stuff.

File: warehouse/models/gtfs_views/gtfs_schedule_dim_shapes.sql [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+WITH shapes_clean AS (
+    SELECT *
+    FROM {{ ref('shapes_clean') }}
+),
+
+gtfs_schedule_dim_shapes AS (
+    SELECT * FROM shapes_clean
+)
+
+SELECT * FROM gtfs_schedule_dim_shapes
```
[the remark this answers is not in the document]
Comment (lauriemerrell) [reply warehouse/models/gtfs_views/gtfs_schedule_dim_shapes.sql:12]: Yeah, this example is maybe a bit much.... It's based on this: https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091. I think I'll leave it for consistency?

File: warehouse/macros/scd_join.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (mjumbewu) [reply warehouse/macros/scd_join.sql:7]: Ah right, I've gotten that before. I believe `none` is lower-cased in Jinja or some suchness. No matter, fine to leave as is for now.

**Pull request merged**